File: app/src/main/java/ch/ralena/natibo/ui/study/overview/StudySessionOverviewFragment.java
package ch.ralena.natibo.ui.study.overview;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import ch.ralena.natibo.ui.MainActivity;
import ch.ralena.natibo.R;
import ch.ralena.natibo.ui.study.insession.adapter.StudySessionAdapter;
import ch.ralena.natibo.data.room.object.Course;
import ch.ralena.natibo.data.room.object.Day;
import ch.ralena.natibo.data.room.object.SentenceGroup;
import ch.ralena.natibo.data.room.object.SentenceSet;
import io.realm.Realm;

public class StudySessionOverviewFragment extends Fragment {
	private static final String TAG = StudySessionOverviewFragment.class.getSimpleName();
	public static final String KEY_COURSE_ID = "language_id";

	Course course;

	private Realm realm;

	private MainActivity activity;

	@Nullable
	@Override
	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		View view = inflater.inflate(R.layout.fragment_study_session_overview, container, false);

		activity = (MainActivity) getActivity();
		activity.setTitle(getString(R.string.session_overview));

		// load schedules from database
		String id = getArguments().getString(KEY_COURSE_ID);
		realm = Realm.getDefaultInstance();
		course = realm.where(Course.class).equalTo("id", id).findFirst();

		// load language name
		TextView courseTitleLabel = view.findViewById(R.id.courseTitleText);
		courseTitleLabel.setText(course.getTitle());

		Day day = course.getCurrentDay();
		List<SentenceGroup> sentenceGroups = new ArrayList<>();
		for (SentenceSet sentenceSet : day.getSentenceSets()) {
			sentenceGroups.addAll(sentenceSet.getSentenceSet());
		}

		// set up recyclerlist and adapter
		RecyclerView recyclerView = view.findViewById(R.id.recyclerView);
		StudySessionAdapter adapter = new StudySessionAdapter(course.getLanguages(), sentenceGroups);
		recyclerView.setAdapter(adapter);
		RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getContext());
		recyclerView.setLayoutManager(layoutManager);

//		adapter.asObservable().subscribe(this::loadSentenceListFragment);

		return view;
	}


}

File: app/src/main/java/ch/ralena/natibo/ui/study/overview/StudySessionOverviewFragment.kt
package ch.ralena.natibo.ui.study.overview

import android.view.View
import ch.ralena.natibo.R
import ch.ralena.natibo.data.room.`object`.CourseRoom
import ch.ralena.natibo.data.room.`object`.LanguageRoom
import ch.ralena.natibo.databinding.FragmentStudySessionOverviewBinding
import ch.ralena.natibo.model.NatiboSession
import ch.ralena.natibo.ui.MainActivity
import ch.ralena.natibo.ui.base.BaseFragment
import ch.ralena.natibo.ui.shared_components.SentenceList
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class StudySessionOverviewFragment :
	BaseFragment<
			FragmentStudySessionOverviewBinding,
			StudySessionOverviewViewModel.Listener,
			StudySessionOverviewViewModel>(FragmentStudySessionOverviewBinding::inflate),
	StudySessionOverviewViewModel.Listener {
	var course: CourseRoom? = null
	private lateinit var activity: MainActivity

	override fun setupViews(view: View) {
		activity = requireActivity() as MainActivity
		activity.title = getString(R.string.session_overview)
		// load schedules from database
		viewModel.registerListener(this)
		val id = requireArguments().getLong(KEY_COURSE_ID)
		viewModel.getSentences(id)
	}

	override fun sessionLoaded(
		session: NatiboSession,
		nativeLanguage: LanguageRoom?,
		targetLanguage: LanguageRoom?
	) {
		binding.composeView.setContent {
			SentenceList(
				sentences = session.sentences,
				nativeLanguage = nativeLanguage,
				targetLanguage = targetLanguage
			)
		}
	}

	companion object {
		private val TAG = StudySessionOverviewFragment::class.java.simpleName
		const val KEY_COURSE_ID = "language_id"
	}
}
